Emby.Server.Implementations/QuickConnect/QuickConnectManager.cs

```diff
@@ -0,0 +1,291 @@
+        /// <param name="logger">Logger.</param>
+        /// <param name="appHost">Application host.</param>
+        /// <param name="authContext">Authentication context.</param>
+        /// <param name="authenticationRepository">Authentication repository.</param>
+        public QuickConnectManager(
+            IServerConfigurationManager config,
+            ILogger<QuickConnectManager> logger,
+            IServerApplicationHost appHost,
+            IAuthorizationContext authContext,
+            IAuthenticationRepository authenticationRepository)
+        {
+            _config = config;
+            _logger = logger;
+            _appHost = appHost;
+            _authContext = authContext;
+            _authenticationRepository = authenticationRepository;
+
+            ReloadConfiguration();
+        }
+
+        /// <inheritdoc/>
+        public int CodeLength { get; set; } = 6;
+
+        /// <inheritdoc/>
+        public string TokenNamePrefix { get; set; } = "QuickConnect-";
+
+        /// <inheritdoc/>
+        public QuickConnectState State { get; private set; } = QuickConnectState.Unavailable;
+
+        /// <inheritdoc/>
+        public int Timeout { get; set; } = 5;
+
+        private DateTime DateActivated { get; set; }
+
+        /// <inheritdoc/>
+        public void AssertActive()
+        {
+            if (State != QuickConnectState.Active)
+            {
+                throw new ArgumentException("Quick connect is not active on this server");
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Activate()
+        {
+            DateActivated = DateTime.Now;
+            SetState(QuickConnectState.Active);
+        }
+
+        /// <inheritdoc/>
+        public void SetState(QuickConnectState newState)
+        {
+            _logger.LogDebug("Changed quick connect state from {State} to {newState}", State, newState);
+
+            ExpireRequests(true);
+
+            State = newState;
+            _config.Configuration.QuickConnectAvailable = newState == QuickConnectState.Available || newState == QuickConnectState.Active;
+            _config.SaveConfiguration();
+
+            _logger.LogDebug("Configuration saved");
+        }
+
+        /// <inheritdoc/>
+        public QuickConnectResult TryConnect(string friendlyName)
+        {
+            ExpireRequests();
+
+            if (State != QuickConnectState.Active)
+            {
+                _logger.LogDebug("Refusing quick connect initiation request, current state is {State}", State);
+                throw new AuthenticationException("Quick connect is not active on this server");
+            }
+
+            _logger.LogDebug("Got new quick connect request from {friendlyName}", friendlyName);
+
+            var code = GenerateCode();
+            var result = new QuickConnectResult()
+            {
+                Secret = GenerateSecureRandom(),
+                FriendlyName = friendlyName,
+                DateAdded = DateTime.Now,
```

Comment:
```suggestion
                DateAdded = DateTime.UtcNow,
```

Let client handle timezone